WalletWasabi/Legal/LegalDocuments.cs

```diff
@@ -10,63 +10,84 @@ namespace WalletWasabi.Legal
 
-		public LegalDocuments(string filePath)
+		public LegalDocuments(Version version, string content)
 		{
-			FilePath = Guard.NotNullOrEmptyOrWhitespace(nameof(filePath), filePath, trim: true);
-			var verString = Path.GetFileNameWithoutExtension(FilePath);
-			Version = Version.Parse(verString);
+			Version = version;
+			Content = content;
 		}
 
-		public string FilePath { get; }
 		public Version Version { get; }
+		public string Content { get; }
 
-		public static LegalDocuments TryLoadAgreed(string dataDir)
+		public static async Task<LegalDocuments?> TryLoadAgreedAsync(string folderPath)
 		{
-			var legalFolderPath = Path.Combine(dataDir, LegalFolderName);
-			IoHelpers.EnsureDirectoryExists(legalFolderPath);
-			var legalDocCandidates = FindCandidates(legalFolderPath);
+			if (!Directory.Exists(folderPath))
+			{
+				return null;
+			}
+
+			var legalDocCandidates = FindCandidates(folderPath);
 			var legalDocCandidateCount = legalDocCandidates.Count();
 
 			if (legalDocCandidateCount == 1)
 			{
 				var filePath = legalDocCandidates.Single();
-				return new LegalDocuments(filePath);
+
+				var content = await File.ReadAllTextAsync(filePath).ConfigureAwait(false);
+				var verString = Path.GetFileNameWithoutExtension(filePath);
+				var version = Version.Parse(verString);
+
+				return new LegalDocuments(version, content);
 			}
 			else if (legalDocCandidateCount > 1)
 			{
-				RemoveCandidates(legalFolderPath);
+				RemoveCandidates(folderPath);
 			}
 
 			return null;
 		}
 
-		private static IEnumerable<string> FindCandidates(string legalFolderPath)
+		private static IEnumerable<string> FindCandidates(string folderPath)
 		{
+			if (!Directory.Exists(folderPath))
+			{
+				return Enumerable.Empty<string>();
+			}
+
 			return Directory
-				.EnumerateFiles(legalFolderPath, "*.txt", SearchOption.TopDirectoryOnly)
+				.EnumerateFiles(folderPath, "*.txt", SearchOption.TopDirectoryOnly)
 				.Where(x => Version.TryParse(Path.GetFileNameWithoutExtension(x), out _));
 		}
 
-		private static void RemoveCandidates(string legalFolderPath)
+		public static void RemoveCandidates(string folderPath)
 		{
-			IoHelpers.EnsureDirectoryExists(legalFolderPath);
-			foreach (var candidate in FindCandidates(legalFolderPath))
+			if (!Directory.Exists(folderPath))
+			{
+				return;
+			}
+
+			foreach (var candidate in FindCandidates(folderPath))
 			{
 				File.Delete(candidate);
 				Logger.LogInfo($"Removed legal doc candidate: {candidate}.");
 			}
 		}
 
-		public async Task ToFileAsync(string content)
+		public async Task ToFileAsync(string folderPath)
 		{
-			var legalFolderPath = Path.GetDirectoryName(FilePath);
-			RemoveCandidates(legalFolderPath);
-			await File.WriteAllTextAsync(FilePath, content).ConfigureAwait(false);
+			folderPath = Guard.NotNullOrEmptyOrWhitespace(nameof(folderPath), folderPath, trim: true);
```

Comment:
This `trim: true` does more damage than good, I think. I don't think that any .NET API actually does this. Mostly it just hides a bug. Also paths with trailing whitespace are not invalid by themselves.

Reply:
I removed it.